src/foreign_calls.rs

```diff
@@ -23,6 +28,36 @@ extern "C" {
     pub type ForeignCallHandler;
 }
 
+pub(super) async fn resolve_brillig(
+    foreign_call_callback: &ForeignCallHandler,
+    foreign_call_wait_info: &ForeignCallWaitInfo,
+) -> Result<ForeignCallResult, String> {
+    // Prepare to call
+    let (name, inputs) = prepare_brillig_args(foreign_call_wait_info);
+
+    // Perform foreign call
+    let outputs = perform_foreign_call(foreign_call_callback, name, inputs).await?;
+    let outputs: Vec<Value> = vecmap(outputs, |output| output.into());
+    // The Brillig VM checks that the number of return values from
+    // the foreign call is valid so we don't need to do it here.
+    Ok(outputs.into())
```

Comment:
This makes the assumption that the foreign call will return a simple array of return values (`Vec<Value>`) whereas it returns a nested array (`Vec<Vec<Value>>`). I've been working on adding this support.